### ISSUE TITLE
Unexpected Large Target Bearing at WP1

### ISSUE THREAD

**Issue by isso11** [ArduPilot/ardupilot#27389]
**Issue details**

This could be a potential issue I've encountered multiple times while using the Software In The Loop (SITL) simulation. During my tests, I observed that upon reaching the first waypoint (WP1), the system consistently outputs a "Large" target bearing value. This occurs despite the fact that the whole mission path is a straight line, and there are no significant bearing errors observed throughout the route.

**Version**
V "4.5.4" commit "e8f937aa"

**Platform**
[  ] All
[  ] AntennaTracker
[  ] Copter
[●] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
Fixed Wing

**Hardware type**
Issue found in SITL (Built-In SITL and flightaxis SITL)

**Logs**
[Log](https://drive.google.com/file/d/1QRGHz5YJZMLDWCmLjKhsr1HstNt53Rab/view?usp=sharing)

**Comment by Georacer** [comment]
Indeed, in the log you provided, the target bearing "jumps" after reaching each waypoint.
However, the actual navigation bearing seems to "jump" only at the first waypoint, bringing along a jump in the roll demand.
![image](https://github.com/ArduPilot/ardupilot/assets/3149996/56e60e11-eb33-4c5f-942a-4d64a1c3e824)

I can't reproduce the issue on my system, though.

Can you provide the SITL command and the mission you use?

**Comment by isso11** [comment]
> Indeed, in the log you provided, the target bearing "jumps" after reaching each waypoint. However, the actual navigation bearing seems to "jump" only at the first waypoint, bringing along a jump in the roll demand. ![image](https://private-user-images.githubusercontent.com/3149996/343163224-56e60e11-eb33-4c5f-942a-4d64a1c3e824.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MDg0ODYsIm5iZiI6MTcxOTQwODE4NiwicGF0aCI6Ii8zMTQ5OTk2LzM0MzE2MzIyNC01NmU2MGUxMS1lYjMzLTRjNWYtOTQyYS00ZDY0YTFjM2U4MjQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjZUMTMyMzA2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2FjYzVmOTYzMDRlZjY1NjM4YjRjMzU3NTYxYzM3N2Q1NTQ4YThiZDFhZGE5NTkwZTM0MTEzZTA5Y2E3OGNlMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.lhGh1YWZSbsEHn0ou1opKreHSFM7IIsHps-I3SBQuGU)
> 
> I can't reproduce the issue on my system, though.
> 
> Can you provide the SITL command and the mission you use?
In the Simulation tab command line: `--home=49.3137086,30.1507759,226.95,22`

Just a trivial mission, Takeoff and then Auto.

Here's the [mission](https://drive.google.com/file/d/1xkiqnbcfT-f1gZwkVNUhIU4sLW1njUud/view?usp=sharing).

**Comment by Georacer** [comment]
Your mission has a "misalignment" of ~1.5deg.
And yes, with this mission it's reproducible in `master`:
![image](https://github.com/ArduPilot/ardupilot/assets/3149996/ac741e47-0182-4fa4-8182-5180e498ef26)

I also modified your mission to "misalign" the segments by ~5 degrees and it still happens, to a reduced degree:
![image](https://github.com/ArduPilot/ardupilot/assets/3149996/15aa2e4f-fa63-4178-9a0c-13075eccf5f6)

Here's the modified mission and the log.
[21217.zip](https://github.com/user-attachments/files/16011964/21217.zip)
[mission_modified.txt](https://github.com/user-attachments/files/16011965/mission_modified.txt)

Looks like a veritable bug to me.
